02-library/cs506/sim.py

```
@@ -1,13 +1,52 @@
+import math
+import decimal
+
+
 def euclidean_dist(x, y):
-    raise NotImplementedError()
+    return (p_root(sum(pow(abs(a - b), 2)for a, b in zip(x, y)), 2))
 
 def manhattan_dist(x, y):
-    raise NotImplementedError()
+    return (p_root(sum(pow(abs(a - b), 1)for a, b in zip(x, y)), 1))
 
 def jaccard_dist(x, y):
-    raise NotImplementedError()
+    if not x and not y:
+        return 1
```

No! distance of 1 implies that the intersection is empty. Please raise an error instead